WooCommerce/Classes/ViewRelated/Themes/ThemeInstaller.swift

```diff
@@ -0,0 +1,128 @@
+    func scheduleThemeInstall(themeID: String, siteID: Int64) {
+        userDefaults.setThemeID(themeID: themeID, for: siteID)
+    }
+
+    /// Installs any pending theme for the given site ID
+    /// - Parameter siteID: site ID to install and activate the theme
+    func installPendingTheme(siteID: Int64) async throws {
+        guard let themeID = userDefaults.themeID(for: siteID) else {
+            return DDLogInfo("No pending theme installation.")
+        }
+
+        DDLogInfo("Attempt to perform pending theme installation. Theme ID: \(themeID), Site ID : \(siteID)")
+        try await installAndActivateTheme(themeID: themeID, siteID: siteID)
+        userDefaults.removeThemeID(for: siteID)
+    }
+}
+
+private extension DefaultThemeInstaller {
+    func installAndActivateTheme(themeID: String, siteID: Int64) async throws {
+        try await installTheme(themeID: themeID, siteID: siteID)
+        try await activateTheme(themeID: themeID, siteID: siteID)
+    }
+
+    @MainActor
+    func installTheme(themeID: String,
+                      siteID: Int64) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(WordPressThemeAction.installTheme(themeID: themeID, siteID: siteID) { result in
+                switch result {
+                case .success:
+                    continuation.resume()
+                case .failure(let error):
+                    if error as? InstallThemeError == .themeAlreadyInstalled {
+                        DDLogInfo("Theme already installed.")
+                        continuation.resume()
+                    } else {
+                        DDLogError("Theme installation failed: \(error)")
+                        continuation.resume(throwing: error)
+                    }
+                }
+            })
+        }
+    }
+
+    @MainActor
+    func activateTheme(themeID: String,
+                       siteID: Int64) async throws {
+        try await withCheckedThrowingContinuation { continuation in
+            stores.dispatch(WordPressThemeAction.activateTheme(themeID: themeID, siteID: siteID) { result in
+                switch result {
+                case .success:
+                    continuation.resume()
+                case .failure(let error):
+                    DDLogError("Theme activation failed: \(error)")
+                    continuation.resume(throwing: error)
+                }
+            })
+        }
+    }
+}
+
+// MARK: - DefaultThemeInstaller helpers
+//
+extension UserDefaults {
+    /// Returns theme ID to be installed for the store
+    ///
+    func themeID(for siteID: Int64) -> String? {
+        let themesPendingInstall = self[.themesPendingInstall] as? [String: String]
+        let idAsString = "\(siteID)"
+        return themesPendingInstall?[idAsString]
+    }
+
+    /// Saves theme ID for installation
+    ///
+    func setThemeID(themeID: String,
```

Comment:
Same as above:

```suggestion
    func setPendingThemeID(_ themeID: String,
```

Reply:
Done as part of e50299450aafc3757c98c13e8d0381d5602a46d1